rollup.config.js

```diff
@@ -4,27 +4,66 @@ import external from 'rollup-plugin-peer-deps-external';
 import postcss from 'rollup-plugin-postcss';
 import resolve from 'rollup-plugin-node-resolve';
 import url from 'rollup-plugin-url';
+
+import glob from 'glob';
+import path from 'node:path';
+import { fileURLToPath } from 'node:url';
+
 const postcssCustomProperties = require('postcss-custom-properties');
 
-export default {
-  input: 'src/lib/index.tsx',
-  output: [
-    {
-      dir: 'dist',
-      format: 'cjs',
-      exports: 'named',
-      sourcemap: true,
-    },
-  ],
-  plugins: [
-    external(),
-    postcss({
-      modules: true,
-      plugins: [postcssCustomProperties({ preserve: true })],
-    }),
-    url(),
-    resolve(),
-    typescript(),
-    commonjs(),
-  ],
-};
+/**
+ *
+ * We need to decide how to isolate cjs and esm builds
+ * For now this totally separates the two builds by defining
+ * explicit input/output properties.
+ *
+ * Need to look into which plugins are needed for each build.
+ *
+ */
+const plugins = [
+  external(),
+  postcss({
+    modules: true,
+    plugins: [postcssCustomProperties({ preserve: true })],
+  }),
+  url(),
+  resolve(),
+  commonjs(),
+];
+
+export default [
+  {
+    input: 'src/lib/index.tsx',
+    output: [
+      {
+        dir: 'dist/cjs',
+        format: 'cjs',
+        exports: 'named',
+        sourcemap: true,
+      },
+    ],
+    plugins: [...plugins, typescript({ outDir: 'dist/cjs' })],
+  },
+  {
+    input: Object.fromEntries(
+      glob
+        // we need to look into why adding ts breaks the build
```

Comment:
Is it maybe picking up test files when adding ".ts"?

Reply:
@kimkwanka  This could actually be the case but also it seems we don't need it at the moment.